vcr/oidc4vci/types.go

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+// This file defines types specified by https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html
+
+package oidc4vci
+
+// PreAuthorizedCodeGrant is the grant type used for pre-authorized code grant from the OIDC4VCI specification.
+// Specified by https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-pre-authorized-code-flow
+const PreAuthorizedCodeGrant = "urn:ietf:params:oauth:grant-type:pre-authorized_code"
+
+// ProviderMetadataWellKnownPath defines the well-known path for retrieving OpenID ProviderMetadata
+// Specified by https://www.rfc-editor.org/rfc/rfc8414.html#section-3
+const ProviderMetadataWellKnownPath = "/.well-known/oauth-authorization-server"
+
+// CredentialIssuerMetadataWellKnownPath defines the well-known path for retrieving OIDC4VCI CredentialIssuerMetadata
+// Specified by https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-credential-issuer-metadata-
+const CredentialIssuerMetadataWellKnownPath = "/.well-known/openid-credential-issuer"
+
+// VerifiableCredentialJSONLDFormat defines the JSON-LD format identifier for Verifiable Credentials.
+const VerifiableCredentialJSONLDFormat = "ldp_vc"
+
+// CredentialIssuerMetadata defines the OIDC4VCI Credential Issuer Metadata.
+// Specified by https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-credential-issuer-metadata
+type CredentialIssuerMetadata struct {
+	// CredentialIssuer defines the identifier of the credential issuer.
+	CredentialIssuer string `json:"credential_issuer"`
+
+	// CredentialEndpoint defines where the wallet can send a request to retrieve a credential.
+	CredentialEndpoint string `json:"credential_endpoint"`
+
+	// CredentialsSupported defines metadata about which credential types the credential issuer can issue.
+	CredentialsSupported []map[string]interface{} `json:"credentials_supported"`
```

Comment:
Anything known about the contents under `interface{}`?

Reply:
Maybe, depends on how evolved we're going to support its configuration. Future work.